### PR TITLE
FORMS-737: Pagination for API calls (submission list endpoint)

### PR DESCRIPTION
**Pull request by bcvesalink**
# Description

Make API calls paginated to improve the performance, start with Get all Submission api call which could have like more than 12000 rows for some forms

## Types of changes

New feature (non-breaking change which adds functionality) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/src/forms/form/service.js [hunk 1]
```diff
@@ -242,14 +242,24 @@
       .modify('filterDeleted', params.deleted)
       .modify('filterCreatedBy', params.createdBy)
       .modify('filterFormVersionId', params.formVersionId)
-      .modify('filterVersion', params.version)
-      .modify('orderDefault');
+      .modify('filterVersion', params.version);
     if (params.createdAt && Array.isArray(params.createdAt) && params.createdAt.length == 2) {
       query.modify('filterCreatedAt', params.createdAt[0], params.createdAt[1]);
     }
+    if (params.search && params.search.length > 2) {
```
[the remark this answers is not in the document]
Comment (WalterMoar) [reply app/src/forms/form/service.js:249]: This looks like a false positive in our case, but it makes me wonder if we should always be checking the type of our parameters (`typeof params.search === 'string'`)

**Comment by WalterMoar** [comment]
On the form submissions table I can't sort by the fields on the form - is this maybe due to that other bug, though? (FORMS-742)

**Comment by jasonchung1871** [comment]
> On the form submissions table I can't sort by the fields on the form - is this maybe due to that other bug, though? (FORMS-742)

I believe this is due to another bug, currently column selection is not working correctly in production.

**Review by jasonchung1871 — APPROVED** [review]

**Review by timisenco2015 — APPROVED** [review]

**Comment by bcvesalink** [comment]
> On the form submissions table I can't sort by the fields on the form - is this maybe due to that other bug, though? (FORMS-742)

@WalterMoar are you talking about additional fields user added to display, those fields are not sortable, sortable only common fields like ConfirmationID, Submission Date, Status, Submitter

**Comment by WalterMoar** [comment]
> > On the form submissions table I can't sort by the fields on the form - is this maybe due to that other bug, though? (FORMS-742)
> 
> @WalterMoar are you talking about additional fields user added to display, those fields are not sortable, sortable only common fields like ConfirmationID, Submission Date, Status, Submitter

In production I can sort by the addition fields from the form that the user added - maybe this isn't supposed to be happening? There is that other bug going on too.

image

**Pull request merged**